File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
import collections

N = int(sys.stdin.readline())

lines = int(sys.stdin.readline())

graph = [[] for i in range(N + 1)]

visited = [False for i in range(N + 1)]

for i in range(lines):
    a, b = map(int, sys.stdin.readline().split(" "))
    graph[a].append(b)
    graph[b].append(a)

count = 0 # 바이러스에 걸린 컴퓨터수

def BFS(start):
    global count
    count += 1
    visited[start] = True
    q = collections.deque([start])
    while q:
        dot = q.popleft()
        for s in graph[dot]:
            if(visited[s] == False):
                visited[s] = True
                count += 1
                q.append(s)


BFS(1)

print(count - 1)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.util.*;
import java.io.*;

public class Main{

    private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    private static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    private static StringTokenizer st;
    private static ArrayList<ArrayList<Integer>> graph = new ArrayList<>();
    private static int answer = 0;
    private static boolean[] visited;

    private static void BFS(int start) {
        Deque<Integer> d = new LinkedList<>();
        d.add(start);
        visited[start] = true;
        while(!d.isEmpty()) {
            int dot = d.poll();
            for(int i = 0 ; i < graph.get(dot).size() ; i++) {
                int newDot = graph.get(dot).get(i);
                if(visited[newDot] == false) {
                    answer += 1;
                    visited[newDot] = true;
                    d.add(newDot);
                }
            }
        }
    }


    public static void main(String[] args) throws IOException {

        int computer = Integer.parseInt(br.readLine());

        int edge = Integer.parseInt(br.readLine());

        visited = new boolean[computer + 1];

        for(int i = 0 ; i <= computer ; i++) {
            graph.add(new ArrayList<>());
        }

        for(int i = 0 ; i < edge ; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            int start = Integer.parseInt(st.nextToken());
            int end = Integer.parseInt(st.nextToken());

            graph.get(start).add(end);
            graph.get(end).add(start);
        }

        BFS(1);

        bw.write(Integer.toString(answer));

        

        bw.flush();

        br.close();
        bw.close();
    }
}